examples/acados_python/getting_started/main_test.py

```diff
@@ -36,34 +36,78 @@
 from random import sample
 
 TEST_SAMPLE = True # only test random sample instaed of all possible combinations
-SAMPLE_SIZE = 20
+SAMPLE_SIZE = 30
```

Comment:
Actually, I am not sure that random testing is a good idea. It would make debugging more difficult in my opinion.

Reply:
CI tests are supposed to run relatively fast.
The combinations of settings are otherwise 486 right now.
It is some kind of trade-off.

Running the 60 test cases takes ~ 1 minute on my machine and I think it should not take much longer.

I agree there might be cases that are not detected right away, but a skipping a module/ setting entirely is very unlikely.

Reply:
Do you have another idea of doing this tradeoff between testing most settings and reasonable runtime?

Reply:
random with fixed seed? then you would get always the same combination at least

Reply:
Then the cases which are not in the "seed" are never tested.
I am not convinced that this is better.

Reply:
Well you said that `skipping a module/ setting entirely is very unlikely`, then we can as well pick an arbitrary set of tests and use it all the time. Otherwise/in addition we can add one test to the Travis matrix that runs all the tests which we turn on only before merging - dunno. I just wouldn't like to be the guy who needs to find a bug with a randomized unit test :p

Reply:
> I just wouldn't like to be the guy who needs to find a bug with a randomized unit test :p

Actually, one can easily turn of the `TEST_SAMPLE` flag or set a fixed combination, which you want to debug when going through the history.

I understand the dislike of the unit test being random though and implemented a fixed seed test in b604241

Reply:
I'm also against random testing, since it could give both "false positive" (i.e. I push something, the test fails, but the issue is not in my push) and "false negative" (i.e. I push something broken but this is not detected). Debugging seems a nightmare.

Many of the fields are orthogonal to each other, so not all combinations of values have to be tested. This greatly reduces the amount of combinations necessary to cover all features.